### PR TITLE
Oklahoma Military Retirement Benefit Exclusion 2022

### LINKED ISSUE #5017: Oklahoma Military Retirement Benefit Exclusion 2022
OK military retirement benefits

[Reference](https://taxsim.nber.org/historical_state_tax_forms/OK/2022/511-Pkt.pdf)

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #5017

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ok/tax/income/agi/subtractions/military_retirement/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+description: Oklahoma limit on military retirement benefit AGI subtraction per person.
```
Comment (hua7450) [inline policyengine_us/parameters/gov/states/ok/tax/income/agi/subtractions/military_retirement/amount.yaml:1]: How should I name this parameter? It's neither a maximum amount nor a minimum amount.

File: policyengine_us/parameters/gov/states/ok/tax/income/agi/subtractions/military_retirement/rate.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+description: Oklahoma excludes this percent of the military retirement benefits.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ok/tax/income/agi/subtractions/military_retirement/rate.yaml:1]:
```suggestion
description: Oklahoma excludes this percentage of military retirement benefits under the retirement benefit subtraction.
```

**Review by PavelMakarchuk — APPROVED** [review]
LGTM after this change

File: policyengine_us/variables/gov/states/ok/tax/income/ok_agi_subtractions.py [hunk 3]
```diff
@@ -18,4 +18,7 @@ def formula(tax_unit, period, parameters):
         pensions = tax_unit.members("taxable_pension_income", period)
         p = parameters(period).gov.states.ok.tax.income.agi.subtractions
         capped_pension = tax_unit.sum(min_(p.pension_limit, pensions))
-        return oasdi + capped_pension
+        military_retirement_exclusion = tax_unit(
+            "ok_military_retirement_exclusion", period
+        )
+        return oasdi + capped_pension + military_retirement_exclusion
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ok/tax/income/ok_agi_subtractions.py:24]: please make this a list parameter, and put the capped pension subtraction into a new variable

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ok/tax/income/subtractions/ok_pension_subtraction.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class ok_pension_subtraction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Oklahoma AGI subtractions from federal AGI"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ok/tax/income/subtractions/ok_pension_subtraction.py:7]:
```suggestion
    label = "Oklahoma pension subtraction"
```

File: policyengine_us/variables/gov/states/ok/tax/income/subtractions/ok_pension_subtraction.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    reference = (
+        "https://oklahoma.gov/content/dam/ok/en/tax/documents/forms/individuals/past-year/2021/511-Pkt-2021.pdf"
+        "https://oklahoma.gov/content/dam/ok/en/tax/documents/forms/individuals/current/511-Pkt.pdf"
+    )
+    defined_for = StateCode.OK
+
+    def formula(tax_unit, period, parameters):
+        pensions = tax_unit.members("taxable_pension_income", period)
+        p = parameters(period).gov.states.ok.tax.income.agi.subtractions
+        capped_pension = tax_unit.sum(min_(p.pension_limit, pensions))
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ok/tax/income/subtractions/ok_pension_subtraction.py:19]: delete next line
```suggestion
        return tax_unit.sum(min_(p.pension_limit, pensions))
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**